File: MEF_examples/ExtendedOperations_CS/ExtendedOperations.cs
using System.ComponentModel.Composition;
using SimpleCalculator;

namespace ExtendedOperations {

  [Export(typeof(IOperation))]
  [ExportMetadata("Symbol", "%")]
  class Modulus : IOperation {
    public int Operate(int left, int right) {
      return left % right;
    }
  };

  [Export(typeof(IOperation))]
  [ExportMetadata("Symbol", "MIN")]
  class Minimum : IOperation {
    public int Operate(int left, int right) {
      return left < right ? left : right;
    }
  };

  [Export(typeof(IOperation))]
  [ExportMetadata("Symbol", "MAX")]
  class Maxumum : IOperation {
    public int Operate(int left, int right) {
      return left > right ? left : right;
    }
  };
}

File: StefAny2_base/StefAny2/ExtendedOperations/ExtendedOperations.cpp
#include "stdafx.h"

namespace ExtendedOperations {

  using namespace System::ComponentModel::Composition;
  //using namespace SimpleCalculator;
  using namespace StefAny2;

  [Export(IOperation::typeid)]
  [ExportMetadata(L"Symbol", L"*")]
  ref class Multiple : IOperation {
  public:
    virtual int Operate(int left, int right) {
      return left * right;
    }
  };

  [Export(IOperation::typeid)]
  [ExportMetadata(L"Symbol", L"/")]
  ref class Devide : IOperation {
  public: 
    virtual int Operate(int left, int right) {
      return left / right;
    }
  };

}
